File: app/src/commonTest/java/in/mvpstarter/sample/common/injection/component/TestComponent.java
package in.mvpstarter.sample.common.injection.component;

import javax.inject.Singleton;

import dagger.Component;
import in.mvpstarter.sample.common.injection.module.ApplicationTestModule;
import in.mvpstarter.sample.injection.component.ApplicationComponent;

@Singleton
@Component(modules = ApplicationTestModule.class)
public interface TestComponent extends ApplicationComponent {

}

File: app/src/commonTest/kotlin/in/mvpstarter/sample/common/injection/component/TestComponent.kt
package `in`.mvpstarter.sample.common.injection.component

import `in`.mvpstarter.sample.common.injection.module.ApplicationTestModule
import `in`.mvpstarter.sample.injection.component.ApplicationComponent
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationTestModule::class))
interface TestComponent : ApplicationComponent